src/main/scala/viper/silver/parser/ParseAst.scala

```diff
@@ -366,6 +367,16 @@ case class PMapType(keyType: PType, valueType: PType)(val pos: (Position, Positi
   override def withTypeArguments(s: Seq[PType]): PMapType = copy(keyType = s.head, valueType = s(1))(pos)
 }
 
+case class PMacroType[T <: PMacro](use: T) extends PType {
```

Comment:
Presumably this is something that should exist only temporarily right after parsing but before macro expansion, right? If so, could you add a comment here that explains this (since it would mean that e.g. the type checker doesn't have to deal with these types any more).